src/modelbench/run_journal.py

```diff
@@ -0,0 +1,128 @@
+import inspect
+import json
+import threading
+from contextlib import AbstractContextManager
+from datetime import datetime, timezone
+from io import IOBase
+from typing import Sequence
+from unittest.mock import MagicMock
+
+from pydantic import BaseModel
+
+from modelbench.benchmark_runner_items import TestRunItem, Timer
+from modelgauge.sut import SUTResponse
+
+
+def for_journal(o):
+    """Turns anything into a collection of primitives suitable for JSON rendering."""
+    if isinstance(o, TestRunItem):
+        return {"test": o.test.uid, "item": o.source_id(), "sut": o.sut.uid}
+    if isinstance(o, SUTResponse):
+        completion = o.completions[0]
```

Comment:
Can we add an assertion here that there is only 1 completion in the response? I'm worried about a potential future scenario in which we ask SUTs for multiple completions and forget to update this code because it's not obvious that the runs are only equipped to handle single-completion responses right now.

Reply:
Is that a plausible scenario? I was thinking that we'd eliminate this along with some of the other unused pluralities.

Reply:
I don't know how likely it is.

Reply:
My general approach to possible futures is to deal with them as they come. Because the number of possible needs is much greater than the number of needs we'll actually have to fill, I think it's much more effective to keep the code simple and be ready to adapt than to build speculatively. Happy to make exceptions for things that we are pretty sure will happen, doing a little extra work to keep our options open, and to deal with things that are low probability but very high risk.

It sounds like in this case we should apply YAGNI, and if it turns out we're wrong we can pay the cost when we know exactly what we do need.

Reply:
Fair enough! Thanks.